python/pyarrow/types.pxi

```diff
@@ -902,6 +902,18 @@ cdef class ExtensionType(BaseExtensionType):
         """
         return ExtensionArray
 
+    def scalar_as_py(self, scalar):
+        """Convert scalar to a Python type.
+
+        This method can be overridden in subclasses to customize what type
+        scalars are converted to.
+
+        Parameters
+        ----------
+        scalar : pyarrow.ExtensionScalar
+          The scalar to be converted to a Python object.
+        """
+        return scalar.as_py() if scalar is not None else None
```

Comment:
This should no longer be able to be None right?

Also is the type ExtensionScalar? It seems we pass the underlying value right?

Reply:
> This should no longer be able to be None right?
>
> Also is the type ExtensionScalar? It seems we pass the underlying value right?

Scalar shouldn't be None, but the return value could I think. Changed, please see if it makes sense.

Reply:
What I meant was, this method is always passed the Scalar of the storage type, right? Not of the Extension type?